src/marvin/beta/assistants/runs.py

```diff
@@ -145,18 +145,16 @@ def _get_tools(self) -> list[AssistantTool]:
             tools.extend(self.additional_tools)
         return tools
 
-    def _get_run_kwargs(self, **run_kwargs) -> dict:
-        if "instructions" not in run_kwargs and (
-            self.instructions is not None or self.additional_instructions is not None
-        ):
-            run_kwargs["instructions"] = self._get_instructions()
-
-        if "tools" not in run_kwargs and (
-            self.tools is not None or self.additional_tools is not None
-        ):
-            run_kwargs["tools"] = self._get_tools()
-        if "model" not in run_kwargs and self.model is not None:
-            run_kwargs["model"] = self._get_model()
+    def _get_run_kwargs(self, thread: Thread = None, **run_kwargs) -> dict:
+        if instructions := self._get_instructions(thread=thread):
+            run_kwargs["instructions"] = instructions
+
+        if tools := self._get_tools():
+            run_kwargs["tools"] = [t.model_dump(mode="json") for t in tools]
```

Comment:
This is the bugfix - we don't let the openai library do the json casting, we do it ourselves